### ISSUE TITLE
Build Call Congress Tool

### ISSUE THREAD

**Issue by simonwex** [MozillaFoundation/plan#263]
We want to connect Americans with their government representatives in order to win on net neutrality.

### RACI
* Phase: Dev / QA / Ship
* Driver: @simonwex *honk* *honk*
* Decision Maker: @valianttry 
* Design Lead: @cassiemc
* Development Lead: @alicoding 
* Dev: @jdotpz 
* Quality Verifier: @simonwex  

### Audience
American activists
### Success
1000 successful callers to congress before **February 26th**.
### Vision
Solution and work can be tracked here:
https://github.com/MozillaFoundation/movement-building/issues/4

**Comment by valianttry** [comment]
Timeline:
* Campaign ends on Feb 26th (U.S. Federal Comms Commission votes on net neutrality that day)
* Mozilla campaign launches February 5th (part of large NGO coalition campaign)
* Calls would ideally be operational from our tool on Feb 5, but as long as it's working no later than Feb 15th that gives us time to push calls before the vote.

**Comment by simonwex** [comment]
Assigning @ScottDowne as lead dev here.

**Comment by simonwex** [comment]
@cassiemc, @edrushka this is a last-minute request, we're trying to shove this in this :heartbeat: as timing is critical. It'd be great to have someone who can do a flyby to help @ScottDowne with design feedback.

**Comment by davidascher** [comment]
I wasn't in the scoping conversation, but I'll add that while 1000 callers may be all we need to call it a win, a) I'm hoping for more (knowing nothing about these things), but b) even if we just hit 1000, having the infrastructure in place to do dial-a-congressman will pay dividends in other campaign.  It's a bit of a foundational piece for our increased impact in advocacy.

**Comment by cassiemc** [comment]
I'm happy to be available for @ScottDowne's design feedback. @flukeout and @vazquez if I can't be reached for some reason.

**Comment by ScottDowne** [comment]
@valianttry @simonwex So question.

This might just be me being not very political, but I'm not really sure what a call congress tool is supposed to do.

I see a enter phone number input, and a "get script and call" button.

What should I expect to happen once I enter my phone number and click that button?

Based on that button, I get a script and I call someone? That's confusing to me.

**Comment by cassiemc** [comment]
I think this needs a kickoff to talk through the scope and UX if that isn't already planned @simonwex @valianttry. I'd love to attend.

**Comment by ScottDowne** [comment]
@cassiemc That would be helpful, most helpful if it can happen today :)

**Comment by simonwex** [comment]
Kickoff scheduled for 13:00 PDT, 16:00 EDT. @ScottDowne we can jam on a couple of the tech bits in the meantime if you've got time.

**Comment by OpenMatt** [comment]
* I'm updating the RACI above.
* Andrea should not be both the owner / project manager AND decider for this work. She's the decider. We need a project manager to  co-ordinate the final technical pieces. 
* Let's make this a priority to assign in Monday TPS meeting. 
* In the mean time, @davidascher and I will PM.

**Comment by OpenMatt** [comment]
* Sub-tasks for this work are here: https://github.com/MozillaFoundation/movement-building/issues

**Comment by OpenMatt** [comment]
* Critical path / schedule being documented here: https://github.com/MozillaFoundation/movement-building/issues/8

**Comment by OpenMatt** [comment]
Updating to Feb13 milestone as this work is ongoing

**Comment by OpenMatt** [comment]
* Priority: we need a clear PM / owner for this going forward.

**Comment by OpenMatt** [comment]
@cassiemc flagging as needs-design -- not sure if design work is complete or ongoing. please update as needed.

**Comment by cassiemc** [comment]
If we want to do Phase 2 w/ the 'report back' feature, yes, needs-design still.

**Comment by OpenMatt** [comment]
@simonwex : are we good to go on this one?

**Comment by simonwex** [comment]
Status update:
 * Secreview is in progress; no issues found yet.
 * Domain `call.mozillafoundation.org` has been selected pending legal approval
 * Front-end code is nearing completion with new design input being incorporated
 * Decision was made to ship tool without attempting to complete P2s.
 * Daily stand-ups are at 10:30 PDT in my vidyo room.

**Comment by davidascher** [comment]
Any ETA on secreview & soonest-available-ship date?

**Comment by jdotpz** [comment]
Secreview is in progress; no issues found yet, looking more today.  cc @ygjb
Now that we've gotten a domain name, I've opened https://bugzilla.mozilla.org/show_bug.cgi?id=1129518 for legal on the call.mozillafoundation.org domain name usage.

We can have this on production infrastructure as soon as we get this domain transferred to us (https://bugzilla.mozilla.org/show_bug.cgi?id=1129466), and legal +1's, and we get ssl.

**Comment by jdotpz** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1129522 is subbed to get people moving.  We really do need EV on this, which makes it rough that we waited until the day of launch to pick a domain name.  This may hose us for launching today.
I've additionally pinged cturra to help

**Comment by davidascher** [comment]
Update:

 - we'll be launching under call.mozilla.org (got ok from jen bertsch and @ygjb).
 - DNS (https://bugzilla.mozilla.org/show_bug.cgi?id=1129674) and SSL (https://bugzilla.mozilla.org/show_bug.cgi?id=1129672) bugs on file

AFAIK, still some design tweaks & QA to go, but at least the domain question is settled.

**Comment by jdotpz** [comment]
All infra and certs are ready.  I have tested the cloudfront hosting, and am turning it off until we go live.  Please let me know at least an hour prior to go-live that we'll be wanting to move on this, and I will re-enable the distribution.

Thanks!

**Comment by simonwex** [comment]
@alicoding has made good progress on the front-end. Still outstanding:
* Speech bubble needs either a design asset or some css-trickery for the top triangle
* We need an appropriately-sized twitter logo
* Copy needs updating
* Form validation needs to happen on the phone number, this is possibly the solution: ( https://github.com/mozilla/call-congress-net-neutrality/issues/2 )

**Comment by cassiemc** [comment]
Speech bubble corner assets are here if it's not reproducable in css: https://www.dropbox.com/sh/67swkuvwhnjq4sz/AAATo830CWem9tPeIESsvBAga?dl=0
Twitter logo is font awesome: http://fortawesome.github.io/Font-Awesome/icon/twitter/

**Comment by cassiemc** [comment]
Twilio logo! That I can do. Coming up.

**Comment by jdotpz** [comment]
@adamlofting has opened https://bugzilla.mozilla.org/show_bug.cgi?id=1129929 via https://github.com/MozillaFoundation/movement-building/issues/11#issuecomment-71997508

**Comment by cassiemc** [comment]
Twillio logo now also here: https://www.dropbox.com/sh/67swkuvwhnjq4sz/AAATo830CWem9tPeIESsvBAga?dl=0 @alicoding @simonwex Was there anything else you needed?

**Comment by simonwex** [comment]
Doing QA, filing boogs here: https://github.com/mozilla/call-congress-net-neutrality/issues/5

**Comment by simonwex** [comment]
@valianttry since we're not enabling the daily-caller functionality, we need to re-record at least the intro. 

The lovely @rubbingalcoholic has offered to re-record the greeting, we need to get him a script. @valianttry can you help write that?

**Comment by valianttry** [comment]
@rubbingalcoholic you are a man of many talents. I assumed we needed to replace this file:
jan14th_msg_intro.mp3

I didn't hear any others that needed to be updated, but please flag any and I'll write the script asap.

Here's a new script:

"Hi this is Jeff on behalf of Mozilla. 
Thanks for making a stand with us to protect net neutrality. Right now every call counts.
Making a call is really easy. During this call, we will connect you to the offices of your 
US Senators and Representative followed by an FCC commissioner in just a second.

When you are connected, give your name and city, so they know you
are their constituent, and say:
"I am calling because the Federal Communications Commissioners will take a vote
February 26th on net neutrality rules. I want the FCC and Congress to do 
everything possible to put strong protections in place. Thank you."

Don't forget to press the star key on your phone after each call instead of hanging up, and you'll
be connected to the next official.

**Comment by jdotpz** [comment]
* During this call, we will connect you to the offices of your 
US Senators and Representative followed by an FCC commissioner in just a second.*

We're actually just calling the FCC this round, afaik.  That's what is in the campaigns.yaml, at least.

Maybe then...
**In just a second, we will connect you to the offices of an FCC commissioner.**

**Comment by valianttry** [comment]
@jdotpz the calls should be to Senate, House, and FCC. That's how the battleforthenet version works, and the call scripts all indicate as much in-call.

**Comment by jdotpz** [comment]
OK, well that is definitely not how it is currently set.

https://github.com/mozilla/call-congress-net-neutrality/blob/master/call-congress.js#L62 is set to use fcc-blanket.  

That is this campaign: https://github.com/mozilla/call-congress/blob/master/data/campaigns.yaml#L315
https://github.com/mozilla/call-congress/blob/master/data/campaigns.yaml#L327 can likely be changed to make that happen.

**Comment by simonwex** [comment]
The current development version can be viewed here: https://mozilla.github.io/call-congress-net-neutrality/

**Comment by adamlofting** [comment]
We're ready to add GA :chart_with_upwards_trend: code now, and I've written a spec within that ticket: https://github.com/MozillaFoundation/movement-building/issues/11

**Comment by jdotpz** [comment]
@valianttry we need a finalized script pretty quick here, think we can get this to @rubbingalcoholic this morning?  @wex and @valianttry we need a final call and testing on the bit to add Congressional calls to this mix, again it would be good to have that in this morning so we can test the flag that turns this on and address issues that pop up.

**Comment by jdotpz** [comment]
Sent Jeff this (removed the 'in just a second' bit since we say 'In a moment' at the beginning of that sentence)

*"Hi this is Jeff on behalf of Mozilla. 
Thanks for making a stand with us to protect net neutrality. Right now every call counts.
Making a call is really easy. During this call, we will connect you to the offices of your 
US Senators and Representative followed by an FCC commissioner in just a second.

When you are connected, give your name and city, so they know you
are their constituent, and say:
"I am calling because the Federal Communications Commissioners will take a vote
February 26th on net neutrality rules. I want the FCC and Congress to do 
everything possible to put strong protections in place. Thank you."

Don't forget to press the star key on your phone after each call instead of hanging up, and you'll
be connected to the next official.*

**Comment by jdotpz** [comment]
Our functionality was different because we did not have the latest code, thus the disconnect on desired functionality vs reality on our hosted version.

We also need to update which campaign we use.  I'll file that.

**Comment by jdotpz** [comment]
We're currently working to resolve this bug, causing the application to tell the caller there was an error after completing the first call.

http://pastebin.mozilla.org/8586621

**Comment by jdotpz** [comment]
Bug resolved, 2nd call feature works beautifully now.

**Comment by OpenMatt** [comment]
* Please update status here before today's Heartbeat Planning meeting. (1pm ET.)
* Also: 11 days since last status update on a P1 is not optimal.

**Comment by ScottDowne** [comment]
I'm not sure what this is blocked on. call.mozilla.org is ready to go and already has some feedback.

Looks like actually getting through to someone is hard. Dropped calls, full voicemail inboxes and being on hold for long waits, but not sure how and if we can fix that.

**Comment by simonwex** [comment]
As @ScottDowne just commented, this is shipped. This passed QA to my satisfaction with the exception of things that are out of our control. @valianttry is happy with where it's at and is planning on pushing traffic to it today.
